service/TransactionsServiceImpl.java

```diff
@@ -8,26 +8,50 @@
 import com.app.utils.Constants;
 import com.app.utils.ValidationUtils;
 import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 import java.time.LocalDateTime;
+import java.time.temporal.TemporalUnit;
 import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
 @Service
 public class TransactionsServiceImpl implements TransactionsService {
 
-    private static final org.slf4j.Logger LOGGER= LoggerFactory.getLogger(TransactionsServiceImpl.class);
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(TransactionsServiceImpl.class);
+    private static final Collection<Transaction> transactions = Collections.synchronizedCollection(new ArrayList<Transaction>());
```

Comment:
What are the disadvantages of Collections.syncrhozedColllection?

Reply:
* That the returned collection does not pass the hashCode and equals operations through to the backing collection, but relies on Object's equals and hashCode methods
and you have to manually synchronize on the returned collection when iterating over it.